### PR TITLE
integrate harmoniously

### LINKED ISSUE #113: Integrate with Harmoniously
Get the honors portion integrated with our current build!

### PR DESCRIPTION
**Pull request by charkour**
This integrates the current app with the honors project. I added the harmoniously tab, but maybe it should be a different page in the top nav bar? I'm open to suggestions.

Currently, you need to upload a schedule and it will infer constraints from that and all users to generate a schedule. It is very crude and ugly, but I intend to update that in multiple future PRs. I'll need to update the API on the harmoniously package before I can update it on our app. 

To test:
- `npm i && npm start`
- Upload a schedule
- Click on `harmoniously` tab
- Click `find schedule`

This PR
- removes deprecated editor settings from the workspace file
- adds a function to get classes on a schedule, and store it in the app context
- adds the harmoniously tab

closes #113

### CODE REVIEW

File: client-course-schedulizer/src/components/Tabs/Harmony/Harmony.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const {
+    appState: { rooms, professors, classes },
+  } = useContext(AppContext);
+
+  const inferredAssignments: Assignments = {};
+  classes.forEach((cls) => {
+    inferredAssignments[cls] = {
+      // TODO: update this. https://stackoverflow.com/a/5915122/9931154
+      professor: professors[Math.floor(Math.random() * professors.length)],
+      rooms: [...rooms, "a", "b"],
```
Comment (Jdellis818) [inline client-course-schedulizer/src/components/Tabs/Harmony/Harmony.tsx:20]: What is the reason for adding "a" and "b" to the list of possible rooms?

**Review by Jdellis818 — APPROVED** [review]
LGTM! It's cool to see your honors project get integrated with the main app. I'm excited to see how you improve upon this as the semester goes on!

File: client-course-schedulizer/src/components/Tabs/Harmony/Harmony.tsx (hunk 1 above)
Comment (charkour) [reply client-course-schedulizer/src/components/Tabs/Harmony/Harmony.tsx:20]: This is just to make sure there are enough rooms to find a schedule, I'll remove it in the future. This is just for testing.

**Comment by charkour** [comment]
Thanks Jon! I'll welcome any feedback on it.

**Pull request merged**